src/lib/framework-config.ts

```diff
@@ -0,0 +1,116 @@
+import type { Integration } from './constants';
+import type { WizardOptions } from '../utils/types';
+
+/**
+ * Configuration interface for framework-specific agent integrations.
+ * Each framework exports a FrameworkConfig that the universal runner uses.
+ */
+export interface FrameworkConfig {
+  metadata: FrameworkMetadata;
+  detection: FrameworkDetection;
+  environment: EnvironmentConfig;
+  analytics: AnalyticsConfig;
+  prompts: PromptConfig;
+  ui: UIConfig;
+}
+
+/**
+ * Basic framework information and documentation
+ */
+export interface FrameworkMetadata {
+  /** Display name (e.g., "Next.js", "React") */
+  name: string;
+
+  /** Integration type from constants */
+  integration: Integration;
+
+  /** URL to framework-specific PostHog docs */
+  docsUrl: string;
+
+  /** Message shown when user declines AI consent */
+  abortMessage: string;
+
+  /**
+   * Optional function to gather framework-specific context before agent runs.
+   * For Next.js: detects router type
+   * For React Native: detects Expo vs bare
+   */
+  gatherContext?: (options: WizardOptions) => Promise<Record<string, any>>;
+}
```

Comment:
Let's hack on this in montreal. I'm concerned about how complicated/bloated this object might get. Also super interested to hash out how this needs to map to docs and stuff.